src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -133,25 +169,28 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 				AddRuntimeLibraries (apk, supportedAbis);
 				apk.Flush();
 				AddNativeLibraries (files, supportedAbis);
-				apk.Flush();
 				AddAdditionalNativeLibraries (files, supportedAbis);
-				apk.Flush();
```

Comment:
I think these two `Flush()` calls were removed on accident?

Reply:
They are not needed as `AddAdditionalNativeLibraries` and the call above do not write to the apk. That is done further down when we look through the `files` list.